crates/katana/rpc/rpc/tests/starknet.rs

```diff
@@ -174,3 +186,201 @@ async fn test_send_declare_and_deploy_legacy_contract() {
+        .expect("Invalid tx receipt");
+
+    let contract_address =
+        get_contract_address(FieldElement::ZERO, res.class_hash, &[], FieldElement::ZERO);
+
+    assert_eq!(
+        contract_address,
+        felt!("0x033d18fcfd3ae75ae4e8a275ce649220ed718b68dc53425b388fedcdbeab5097")
+    );
+
+    // Messaging between L1 -> L2
+    let builder = contract_c1
+        .sendMessage(
+            U256::from_str("0x033d18fcfd3ae75ae4e8a275ce649220ed718b68dc53425b388fedcdbeab5097")
+                .unwrap(),
+            U256::from_str("0x005421de947699472df434466845d68528f221a52fce7ad2934c5dae2e1f1cdc")
+                .unwrap(),
+            vec![U256::from(123)],
+        )
+        .gas(12000000)
+        .value(Uint::from(1));
+
+    let receipt = builder
+        .send()
+        .await
+        .expect("Error Await pending transaction")
+        .get_receipt()
+        .await
+        .expect("Error getting transaction receipt");
+
+    assert!(receipt.status());
+
+    // wait for the tx to be mined (Using delay cause the transaction is sent from L1 and is
+    // received in L2)
+    tokio::time::sleep(Duration::from_millis(WAIT_TX_DELAY_MILLIS)).await;
+
+    let tx = starknet_account
+        .provider()
+        .get_transaction_by_block_id_and_index(BlockId::Tag(BlockTag::Latest), 0)
+        .await
+        .unwrap();
+
+    match tx {
+        Transaction::L1Handler(ref l1_handler_transaction) => {
+            let calldata = &l1_handler_transaction.calldata;
+
+            assert_eq!(
+                tx.transaction_hash(),
+                &felt!("0x00c33cc113afc56bc878034908472770cb13eda6ad8ad91feb25fd4e5c9196a0")
+            );
+
+            assert_eq!(FieldElement::to_string(&calldata[1]), "123")
+        }
+        _ => {
+            panic!("Error, No L1handler transaction")
+        }
+    }
+
+    // Messaging between L2 -> L1
+    let cairo_messaging_contract = CairoMessagingContract::new(contract_address, &starknet_account);
+    let tx = cairo_messaging_contract
+        .send_message_value(
+            &EthAddress::from(
+                FieldElement::from_str(contract_c1.address().to_string().as_str()).unwrap(),
+            ),
+            &FieldElement::from(2u8),
+        )
+        .send()
+        .await
+        .expect("Call to send_message_value failed");
+
+    TransactionWaiter::new(tx.transaction_hash, starknet_account.provider())
+        .with_tx_status(TransactionFinalityStatus::AcceptedOnL2)
+        .await
+        .expect("Invalid tx receipt");
+
+    let builder = contract_c1
+        .consumeMessage(
+            U256::from_str(contract_address.to_string().as_str()).unwrap(),
+            vec![U256::from(2)],
+        )
+        .value(Uint::from(1))
+        .gas(12000000)
+        .nonce(4);
+
+    // Wait for the message to reach L1
+    tokio::time::sleep(Duration::from_millis(8000)).await;
+
+    let tx_receipt = builder.send().await.unwrap().get_receipt().await.unwrap();
```

Comment:
@kariy @glihm The assert of the tx_receipt says status: False, do not know what I am doing wrong here. This is the only part missing to complete messaging between L2 -> L1.